### ISSUE TITLE
 ./test.sh failing 

### ISSUE THREAD

**Issue by odidev** [hauntsaninja/boostedblob#9]
I have been working on installing and testing boostedblob package for x86_64 and arm64 architectures on my local server. While testing this package using **./test.sh** I am getting the below error. It would be really helpful if you could share some pointers on it: 
``` 
root@38e33a01fc7f:/boostedblob# ./test.sh 
++ black --diff --check --quiet . 
++ isort --diff --check --quiet . 
++ flake8 
++ mypy boostedblob 
boostedblob/cli.py:23: error: unused "type: ignore" comment 
Found 1 error in 1 file (checked 16 source files) 
```

**Comment by hauntsaninja** [comment]
Thanks! Without that type-ignore, I get `boostedblob/cli.py:23: error: Module has no attribute "__version__"` (using mypy 0.910 and uvloop 0.16.0, as specified in pyproject.toml).
mypy gives this error because a) mypy.ini specifies `strict = True`, so `implicit_reexport = False`, b) `uvloop` ships with py.typed (as of v0.15), and does not include `__version__` in `uvloop.__init__.__all__`.
poetry should ensure you have the right versions, but is it possible you have uvloop 0.14?

Note also that unit tests may be tricky for you to run, since you will not have access to the default test locations. See https://github.com/hauntsaninja/boostedblob/blob/master/CONTRIBUTING.md#running-tests for details on that.

**Comment by odidev** [comment]
@hauntsaninja thanks for quick response, 
To run unit tests is it possible that I get access to the default test locations? 

After installing uvloop 0.14 using ```pip install uvloop==0.14``` I am getting this error ```boostedblob 0.9.2 has requirement uvloop<0.17.0,>=0.16.0, but you'll have uvloop 0.14.0 which is incompatible.```

 
Without type-ignore I get: 
```
root@38e33a01fc7f:/boostedblob# ./test.sh 

++ black --diff --check --quiet . 

--- boostedblob/cli.py  2021-12-15 08:05:09.691052 +0000 

+++ boostedblob/cli.py  2021-12-15 08:09:38.654008 +0000 

@@ -18,11 +18,11 @@ 

     def wrapper(*args: Any, **kwargs: Any) -> Any: 

         try: 

             import uvloop 

  

             if sys.version_info < (3, 9) or ( 

-                tuple(map(int, uvloop.__version__.split("."))) >= (0, 15, 0) 

+                tuple(map(int, uvloop.__version__.split("."))) >= (0, 15, 0) 

             ): 

                 uvloop.install() 

         except ImportError: 

             pass 

         return asyncio.run(fn(*args, **kwargs)) 
```

**Comment by hauntsaninja** [comment]
uvloop 0.16 is the correct version to be using. I mentioned 0.14 as a possible hypothesis for why we were seeing different mypy results.

The diff you posted is from black failing: https://github.com/hauntsaninja/boostedblob/blob/256c5acf039a8f5ef0b799ac42749f7e4ab11da5/test.sh#L3
I assume this is because of a trailing space or something left when removing the type ignore. Because of the `set -ex`, this meant that `mypy boostedblob` was never even run.

If you have the correct uvloop version, another possibility for why you're getting different results with `mypy boostedblob` is that mypy isn't installed in the same virtual environment as uvloop (and so due to `--ignore-missing-imports`, `uvloop` gets replaced by `Any`). Thanks to poetry, my environment should be easily reproducible: just run `poetry install`, `poetry shell`, and see if it still fails.

Unfortunately, I can't provide access to the default test locations, since they're tied to my work.

**Comment by odidev** [comment]
@hauntsaninja thanks for response,

I treid the below commands to install the dependencies on x86 and arm64 architectures: 
- pip3 install boostedblob   
- curl -sSL https://install.python-poetry.org | python3 - 
- pip3 install poetry 
- poetry install  
- poetry shell 

To ran the test I am using ```./test.sh``` Without that ```type-ignore``` and with ```uvloop 0.16.0``` I am getting the below error.  
```
(boostedblob-2qfDnXVF-py3.8) root@62b6e46f8b68:/boostedblob# ./test.sh 

++ black --diff --check --quiet . 

++ isort --diff --check --quiet . 

++ flake8 

++ mypy boostedblob 

Success: no issues found in 16 source files 

++ mypy tests --disable-error-code var-annotated 

Success: no issues found in 16 source files 

++ [[ -z '' ]] 

++ export BBB_TEST_LOCATIONS=az://shantanutest/container 

++ BBB_TEST_LOCATIONS=az://shantanutest/container 

++ pytest --cov=boostedblob 

ImportError while loading conftest '/boostedblob/tests/conftest.py'. 

tests/conftest.py:1: in <module> 

    from .helpers import any_dir, other_any_dir 

tests/helpers/__init__.py:1: in <module> 

    from .create import * 

tests/helpers/create.py:14: in <module> 

    def unsafe_create_file(path: BasePath, contents: bytes = b"asdf") -> asyncio.Task[None]: 

E   TypeError: 'type' object is not subscriptable 
```
Could you please provide your feddback on this?

**Comment by hauntsaninja** [comment]
Aha! Okay, using Python 3.7 explains both these errors and the mypy errors previously mentioned. I just pushed a fix in fb730652e294c05f2d195fc67503d78aab785ee6

I believe now if you checkout master and run poetry install:
```
BBB_TEST_LOCATIONS='' ./test.sh
```
should pass for you.

The reason you need to pass that environment variable is because you don't have access to the default testing locations; so it skips all tests that would require access to those location. If you do have access to some other Azure Blob storage account and/or Google Cloud storage bucket, you can pass their value(s) into `BBB_TEST_LOCATIONS` to test against them (as mentioned in CONTRIBUTING.md).

**Comment by hauntsaninja** [comment]
Closing, since I believe ^ fixes issues discussed here. Let me know if you still have trouble :-)
